File: python/questions/minPairSum.py
class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        l = 0
        r = len(nums) - 1
        nums.sort()
        ans = -inf
        while l < r:
            ans = max(ans, nums[l]+nums[r])
            l += 1
            r -= 1
        return ans

File: javascript/questions/minPairSum.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minPairSum = function(nums) {
    nums.sort((a, b) => a - b);
    let l = 0;
    let r = nums.length - 1;
    let ans = -Infinity;
    while (l < r) {
        const cps = nums[l] + nums[r];
        if (ans < cps) {
            ans = cps;
        }
        l++;
        r--;
    }
    return ans;
};
